src/Sarif/Core/Rule.cs

```diff
@@ -13,5 +13,12 @@ public string Format(string messageId, IEnumerable<string> arguments)
         {
             return string.Format(CultureInfo.CurrentCulture, this.MessageStrings[messageId], arguments.ToArray());
         }
+
+        public bool ShouldSerializeDeprecatedIds()
+        {
+            return this.DeprecatedIds != null &&
+                (this.DeprecatedIds.Where((e) => { return e != null; }).Count() == this.DeprecatedIds.Count);
```

Comment:
>Where [](start = 36, length = 5)

`Count` takes a predicate (and my goodness, so many extra keystrokes in that lambda :-)):
```
this.DeprecatedIds.Count(e => e != null) == this.DeprecatedIds.Count()
```

Reply:
But is this the right test? You could just
```
return this.DeprecatedIds?.Count(e => e != null) > 0;
```
... and then persist the non-`null` ones.

---
In reply to: [249160403](https://github.com/Microsoft/sarif-sdk/pull/1223#discussion_r249160403) [](ancestors = 249160403)

Reply:
yes, greatly  reduced, like it.

---
In reply to: [249182441](https://github.com/Microsoft/sarif-sdk/pull/1223#discussion_r249182441) [](ancestors = 249182441,249160403)